hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieAvroDataBlock.java

```diff
@@ -153,11 +156,14 @@ protected <T> ClosableIterator<T> deserializeRecords(HoodieReaderContext<T> read
   }
 
   private static class RecordIterator implements ClosableIterator<IndexedRecord> {
+
+    private final Boolean whichImplementation = false;
     private byte[] content;
     private final SizeAwareDataInputStream dis;
     private final GenericDatumReader<IndexedRecord> reader;
     private final ThreadLocal<BinaryDecoder> decoderCache = new ThreadLocal<>();
-
+    private Option<Schema> castSchema = Option.empty();
+    private UnaryOperator<Object> converter;
```

Comment:
My high-level question here is why does data block need to concerned about schema evolution. This should just get the evolved schema from a layer above isn't it?

Reply:
[Avro schema type promotion](https://avro.apache.org/docs/1.11.1/specification/) (just ctrl-f for promotion in the link) does not accommodate promotion from number types to string. This is a common ask so we are adding the feature.